src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -540,9 +553,14 @@ int MPL_gavl_tree_delete(MPL_gavl_tree_t gavl_tree, const void *addr, uintptr_t
             goto fn_exit;
         }
 
-        /* delete the matched node and free corresponding buffer object val */
+        /* detach the matched node from tree */
         gavl_tree_delete_internal(tree_ptr, dnode);
 
```

Comment:
I actually do not see why the new version is better than the original one. In the new version, you have to update so many pointers (e.g., parent, left, right, and their pointers).

Reply:
If we keep the original version, the function would become `gavl_tree_delete_internal(tree_ptr, &dnode);`
we have `&dnode` because `inorder_node` could be deleted instead of `dnode` and returned to be added into `removed_list`.

This design makes confusion that why dnode will be changed during `gavl_tree_delete_internal` and this is not straightforward. So I decided to use this new design.

Reply:
The latest code moves `dnode` into the `remove_list` inside `gavl_tree_remove_node_internal`. Do we still have the above concern? I'd prefer the original simpler code because the new version involves many pointer updates which are not easy to maintain.

Reply:
Yes, you are right. My bad, I miss our new changes here :) 
I will fix it.